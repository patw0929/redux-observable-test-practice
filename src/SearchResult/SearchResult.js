import React from 'react';

const SearchResult = ({ items }) => (
  <ul className="list-group">
    {items.map(item => (
      <li className="list-group-item">{item.full_name}</li>
    ))}
  </ul>
);

export default SearchResult;
