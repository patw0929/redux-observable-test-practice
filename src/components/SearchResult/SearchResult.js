import React, { Fragment } from 'react';
import Spinner from '../Spinner/Spinner';
import ErrorMessage from '../ErrorMessage/ErrorMessage';

const SearchResult = ({ items, isFetching, error }) => {
  if (isFetching) {
    return <Spinner />;
  }

  return (
    <Fragment>
      {!!error && <ErrorMessage text={error} />}
      <ul className="list-group">
        {items.map(item => (
          <li
            key={item.id}
            className="list-group-item"
          >
            {item.full_name}
          </li>
        ))}
      </ul>
    </Fragment>
  );
};

export default SearchResult;
