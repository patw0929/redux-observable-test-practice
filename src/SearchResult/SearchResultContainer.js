import { connect } from 'react-redux';
import SearchResult from './SearchResult';

const mapStateToProps = state => {
  if (!state.keyword) {
    return {
      items: [],
    };
  }

  return {
    items: state.searchByKeyword[state.keyword].items,
  };
};

export default connect(mapStateToProps)(SearchResult);
