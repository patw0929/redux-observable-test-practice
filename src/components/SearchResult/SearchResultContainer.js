import { connect } from 'react-redux';
import SearchResult from './SearchResult';

const mapStateToProps = state => {
  const isFetching = state.searchByKeyword.isFetching;

  if (!state.searchByKeyword.keyword) {
    return {
      isFetching: false,
      items: [],
    };
  }

  return {
    isFetching,
    items: state.searchByKeyword.keywords[state.searchByKeyword.keyword].items,
  };
};

export default connect(mapStateToProps)(SearchResult);
