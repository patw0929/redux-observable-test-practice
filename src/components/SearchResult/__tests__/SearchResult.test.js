import React from 'react';
import { shallow } from 'enzyme';
import SearchResult from '../SearchResult';

describe('SearchResult', () => {
  it('renders correct as snapshot - with isFetching is true', () => {
    const subject = shallow(<SearchResult isFetching />);

    expect(subject.html()).toMatchSnapshot();
  });

  it('renders correct as snapshot - with has error', () => {
    const subject = shallow(
      <SearchResult
        isFetching={false}
        items={[]}
        error="Network error!"
      />
    );

    expect(subject.html()).toMatchSnapshot();
  });

  it('renders correct as snapshot - with has items', () => {
    const subject = shallow(
      <SearchResult
        isFetching={false}
        items={[{
          id: 1,
          full_name: 'react',
        }, {
          id: 2,
          full_name: 'redux',
        }]}
        error={null}
      />
    );

    expect(subject.html()).toMatchSnapshot();
  });
});
