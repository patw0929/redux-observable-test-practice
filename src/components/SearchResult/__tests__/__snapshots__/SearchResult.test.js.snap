// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult renders correct as snapshot - with has error 1`] = `"<div class=\\"alert alert-danger\\" role=\\"alert\\"><strong>Oops!</strong> Network error!</div><ul class=\\"list-group\\"></ul>"`;

exports[`SearchResult renders correct as snapshot - with has items 1`] = `"<ul class=\\"list-group\\"><li class=\\"list-group-item\\">react</li><li class=\\"list-group-item\\">redux</li></ul>"`;

exports[`SearchResult renders correct as snapshot - with isFetching is true 1`] = `"<div class=\\"loader\\">Loading...</div>"`;
